sources/packages/backend/libs/services/src/students-assessments/assessment-sequential-processing.service.ts

```diff
@@ -92,6 +105,124 @@ export class AssessmentSequentialProcessingService {
+    });
+    const sequencedApplications = await this.getSequencedApplications(
+      assessment.application.applicationNumber,
+      assessment.application.student.id,
+      assessment.application.programYear.id,
+      { alternativeReferenceDate: options?.alternativeReferenceDate },
+    );
+    if (!sequencedApplications.previous.length) {
+      // There are no past applications.
+      return [];
+    }
+    const applicationNumbers = sequencedApplications.previous.map(
+      (application) => application.applicationNumber,
+    );
+    const totals = await this.studentAssessmentRepo
+      .createQueryBuilder("studentAssessment")
+      .select("disbursementValue.valueCode", "valueCode")
+      .addSelect("offering.offeringIntensity", "offeringIntensity")
+      .addSelect("SUM(disbursementValue.valueAmount)", "total")
+      .innerJoin("studentAssessment.application", "application")
+      .innerJoin(
+        "studentAssessment.disbursementSchedules",
+        "disbursementSchedule",
+      )
+      .innerJoin("disbursementSchedule.disbursementValues", "disbursementValue")
+      .innerJoin("studentAssessment.offering", "offering")
+      .where("application.applicationNumber IN (:...applicationNumbers)", {
+        applicationNumbers,
+      })
+      // Overwritten application can have awards associated with and they should not be considered.
+      .andWhere("application.applicationStatus != :overwrittenStatus", {
+        overwrittenStatus: ApplicationStatus.Overwritten,
+      })
+      // Check for assessment completed status to avoid retrieving any cancelation status.
+      .andWhere(
+        "studentAssessment.studentAssessmentStatus = :completedStudentAssessmentStatus",
+        {
+          completedStudentAssessmentStatus: StudentAssessmentStatus.Completed,
+        },
+      )
+      // Only consider disbursements in Pending, ReadyToSend, or Sent.
+      .andWhere(
+        "disbursementSchedule.disbursementScheduleStatus != :cancelledDisbursementScheduleStatus",
+        {
+          cancelledDisbursementScheduleStatus:
+            DisbursementScheduleStatus.Cancelled,
+        },
+      )
+      // Sequenced applications need at least one valid COE, which means that one can be cancelled
+      // and the other still valid. This ensures that only the valid one will be considered.
+      .andWhere("disbursementSchedule.coeStatus != :declinedCOEStatus", {
+        declinedCOEStatus: COEStatus.declined,
+      })
+      // Ensures that only grants will be returned since loans and not needed.
+      .andWhere("disbursementValue.valueType IN (:...grantsValueTypes)", {
+        grantsValueTypes: [
+          DisbursementValueType.CanadaGrant,
+          DisbursementValueType.BCGrant,
+        ],
+      })
+      .groupBy("offering.offeringIntensity")
+      .addGroupBy("disbursementValue.valueCode")
+      .getRawMany<{
+        offeringIntensity: OfferingIntensity;
+        valueCode: string;
+        total: string;
```

Comment:
Shouldn't  it be integer?

Reply:
The `getRawMany` returns the values from DB without any conversion but since the numeric range cannot be handled by javascript `number` type the DB driver will have it as `string`.
Changing the type from the `getRawMany` will not change it.
![image](https://github.com/bcgov/SIMS/assets/61259237/eba3752d-3d78-48ff-9367-261f2bf4e805)